### DISCUSSION TITLE
How to create a custom page i.e, changepassword page using livewire in filament dashboard

### DISCUSSION THREAD

**Discussion by mohandtharsha** [filamentphp/filament#8170] [Help]
I want to create custom page using livewire in filament dashboard

**Comment by leandrocfe** [comment]
https://filamentphp.com/docs/3.x/panels/pages/#creating-a-page

Pages are livewire components

**Reply by mohandtharsha** [reply]
I have got that But I am not able to see the submit button for edit form below is my code, I want to show update button update the data


<?php

namespace App\Filament\Employer\Pages;
use App\Filament\Employer\Resources\JobsResource\RelationManagers;
use Filament\Forms\Concerns\InteractsWithForms;
use Filament\Pages\Page;
use Filament\Forms;
use Filament\Forms\Form;
use Filament\Forms\Components\TextInput;
use App\Models\Users;
use Filament\Forms\Contracts\HasForms;
use Filament\Tables\Columns\TextColumn;
use Filament\Forms\Components\FileUpload;
use Filament\Forms\Components\Select;
use Filament\Tables\Columns\ImageColumn;
use Filament\Forms\Components\SpatieMediaLibraryFileUpload;
use Filament\Tables\Columns\SpatieMediaLibraryImageColumn;
use Filament\Forms\Components\Card;
use Filament\Forms\Components\MarkdownEditor;
use App\Models\EmployerPersonalInfo;
use Filament\Actions;
use Filament\Infolists\Components\TextEntry;
use Illuminate\Contracts\View\View;
use Illuminate\Support\HtmlString;

class EmployersProfile extends Page implements HasForms
{
    use InteractsWithForms;

    public EmployerPersonalInfo $post;

    protected static ?string $model = EmployerPersonalInfo::class;

    protected static ?string $navigationIcon = 'heroicon-o-document-text';

    protected static string $view = 'filament.employer.pages.employers-profile';
    public $about_the_organization;
    public $website;
    public $organization_logo;
    public $type_of_organization;
    public $category_or_sector;

    public function __construct()
    {
        $this->personalInfo = new EmployerPersonalInfo(); // Initialize the $post property
    }

    public function mount(): void
    {
        // Retrieve an existing EmployerPersonalInfo record from the database by ID
        $this->personalInfo = EmployerPersonalInfo::find(1);
        // Check if a record was found
        if ($this->personalInfo) {
            // Fill the form with the retrieved data
            $this->form->fill([
                'about_the_organization' => $this->personalInfo->about_the_organization,
                'website' => $this->personalInfo->website,
                'organization_logo' => $this->personalInfo->organization_logo,
                'type_of_organization' => $this->personalInfo->type_of_organization,
                'category_or_sector' => $this->personalInfo->category_or_sector,
                // Other fields
            ]);
        }
    }

    protected function getFormSchema(): array
    {
        return [
                MarkdownEditor::make('about_the_organization')
                    ->label('About the organization')
                    ->required(),
                TextInput::make('website')
                    ->label('Website')
                    ->required(),
                SpatieMediaLibraryFileUpload::make('organization_logo')
                ->label('Logo'),
                Select::make('category_or_sector')
                            // ->required()
                            ->options(\App\Models\Categories::pluck('name', 'id'))
                            ->label('Category or Sector of the Organization'),

                Select::make('type_of_organization')
                    ->label('Organization Type')
                    ->required()
                    ->options([
                        'private' => 'Private',
                        'public' => 'Public',
                    ]),
        ];
    }


    public function save(): void
    {
        $this->personalInfo->update(
            $this->form->getState(),
        );
    }
}

**Reply by leandrocfe** [reply]
https://filamentphp.com/docs/3.x/forms/adding-a-form-to-a-livewire-component#adding-the-form

**Reply by mohandtharsha** [reply]
I am getting below error,

Unable to call component method. Public method [create] not found on component

**Reply by leandrocfe** [reply]
share the whole code please... php class and blade file

**Reply by mohandtharsha** [reply]
if you know the answer please send the code dont simply give links

**Reply by krekas** [reply]
please first learn livewire. and if you are posting code please format it using markdown. if you don't know how google it.
